src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,314 @@
+  text = extractPattern(text, TYPE_EXPRESSION, 'type');
+  text = extractPattern(text, DEFAULT_EXPRESSION, 'default');
+
+  // Assign the remaining text as the description after removing leading hyphens
+  current.desc = text.replace(LEADING_HYPHEN, '').trim() || undefined;
+
+  // Recursively parse nested options if a list is found within the list item
+  const optionsNode = child.children.find(child => child.type === 'list');
+  if (optionsNode) {
+    current.options = optionsNode.children.map(child =>
+      parseListItem(child, entry)
+    );
+  }
+
+  return current;
+}
+
+/**
+ * Parses stability metadata and adds it to the section.
+ * @param {import('../types.d.ts').Section} section - The section to add stability to.
+ * @param {Array} nodes - The AST nodes.
+ */
+function parseStability(section, nodes) {
+  nodes.forEach((node, i) => {
+    if (
+      node.type === 'blockquote' &&
+      node.children.length === 1 &&
+      node.children[0].type === 'paragraph' &&
+      nodes.slice(0, i).every(n => n.type === 'list')
+    ) {
+      const text = transformNodesToString(node.children[0].children);
+      const stabilityMatch = /^Stability: ([0-5])(?:\s*-\s*)?(.*)$/s.exec(text);
+      if (stabilityMatch) {
+        section.stability = Number(stabilityMatch[1]);
+        section.stabilityText = stabilityMatch[2].replace(/\n/g, ' ').trim();
+        nodes.splice(i, 1); // Remove the matched stability node to prevent further processing
+      }
+    }
+  });
+}
+
+/**
+ * Parses a list and updates the section accordingly.
+ * @param {import('../types.d.ts').Section} section - The section to update.
+ * @param {Array} nodes - The AST nodes.
+ * @param {import('../types.d.ts').HierarchizedEntry} entry - The associated entry.
+ */
+function parseList(section, nodes, entry) {
+  const list = nodes[0]?.type === 'list' ? nodes.shift() : null;
+  const values = list
+    ? list.children.map(child => parseListItem(child, entry))
+    : [];
+
+  switch (section.type) {
+    case 'ctor':
+    case 'classMethod':
+    case 'method':
+      section.signatures = [parseSignature(section.textRaw, values)];
+      break;
+    case 'property':
+      if (values.length) {
+        const { type, ...rest } = values[0];
+        if (type) section.propertySigType = type;
+        Object.assign(section, rest);
+        section.textRaw = `\`${section.name}\` ${section.textRaw}`;
+      }
+      break;
+    case 'event':
+      section.params = values;
+      break;
+    default:
+      if (list) nodes.unshift(list); // If the list wasn't processed, add it back for further processing
+  }
+}
+
+/**
+ * Adds a description to the section.
+ * @param {import('../types.d.ts').Section} section - The section to add description to.
+ * @param {Array} nodes - The AST nodes.
+ */
+function addDescription(section, nodes) {
+  if (!nodes.length) return;
+
+  if (section.desc) {
+    section.shortDesc = section.desc;
+  }
+
+  const html = getRemarkRehype();
```

Comment:
I feel we would be calling remark-rehype one time per each description right? Which means initialising remark-rehpye multiple times.

I wonder if you could simply grab all description sections and parse at once then splice the nodes and add them badck.